mlir/lib/Target/SPIRV/SPIRVBinaryUtils.cpp

```diff
@@ -51,7 +51,7 @@ void spirv::appendModuleHeader(SmallVectorImpl<uint32_t> &header,
   // | 0 (reserved for instruction schema)                                     |
   // +-------------------------------------------------------------------------+
   header.push_back(spirv::kMagicNumber);
-  header.push_back((majorVersion << 16) | (minorVersion << 8));
+  header.push_back((majorVersion << 8) | minorVersion);
   header.push_back(kGeneratorNumber);
```

Comment:
The generator number was the thing the issue was about, and that isn't changed here, so this pull request doesn't fix the issue.